crates/re_types/src/lib.rs

```diff
@@ -341,25 +351,86 @@ pub enum DeserializationError {
     OffsetsMismatch {
         bounds: (usize, usize),
         len: usize,
-        datatype: ::arrow2::datatypes::DataType,
+        backtrace: _Backtrace,
     },
 
     #[error("Expected array of length {expected} but found a length of {got:#?} instead")]
     ArrayLengthMismatch {
         expected: usize,
         got: usize,
-        datatype: ::arrow2::datatypes::DataType,
+        backtrace: _Backtrace,
     },
 
     #[error("Expected single-instanced component but found {got} instances instead")]
-    MonoMismatch {
-        got: usize,
-        datatype: ::arrow2::datatypes::DataType,
-    },
+    MonoMismatch { got: usize, backtrace: _Backtrace },
 }
 
 pub type DeserializationResult<T> = ::std::result::Result<T, DeserializationError>;
 
+trait ResultExt<T> {
+    fn detailed_unwrap(self) -> T;
+}
+
+impl<T> ResultExt<T> for SerializationResult<T> {
+    fn detailed_unwrap(self) -> T {
+        fn find_backtrace(_err: &SerializationError) -> Option<_Backtrace> {
+            None
+        }
```

Comment:
I see why this is here despite being no-op atm, but to force it to fullfill its destiny, shouldn't you do a match on `SerializationError` so that someone adding a new one will be forced to fill out the new arms that may lead to a backtrace?